### PR TITLE
fix: ga4 issues

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change
- identify call support removed
- login, sign_up and generate_lead events are added as track events
- clientId is mapped with anonymousId
- unnecessary test cases removed

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by shrouti1507 — COMMENTED** [review]
@mihir-4116 did you consider [this](https://github.com/rudderlabs/rudder-transformer/pull/1845) PR, as this is not yet merged. Needed to confirm.

**Comment by mihir-4116** [comment]
> @mihir-4116 did you consider [this](https://github.com/rudderlabs/rudder-transformer/pull/1845) PR, as this is not yet merged. Needed to confirm.

yes @shrouti1507 , both changes are different so won't affect each other

File: src/v0/destinations/ga4/transform.js [hunk 1]
```diff
@@ -129,114 +109,49 @@ const responseBuilder = (message, { Config }) => {
...
-    // for select_item and view_item event we take custom properties from properties
-    // excluding items/product properties
-    if (payload.name === 'select_item' || payload.name === 'view_item') {
-      // exclude event properties
-      let ITEM_EXCLUSION_LIST = getGA4ExclusionList(mappingConfig[name]);
-      // exclude items/product properties
-      ITEM_EXCLUSION_LIST = ITEM_EXCLUSION_LIST.concat(
-        getGA4ExclusionList(mappingConfig[ConfigCategory.ITEM.name]),
-      );
+    if (!USER_IDENTIFIERS_EVENTS.includes(payload.name)) {
```
Comment (utsabc) [inline src/v0/destinations/ga4/transform.js:112]: Where are we handling the `login`, `sign_up`, `generate_lead` event?

Comment (mihir-4116) [reply src/v0/destinations/ga4/transform.js]: from mapping config

`payload.params = constructPayload(message, mappingConfig[name]);`

all 3 event has mapping config

File: src/v0/destinations/ga4/transform.js [hunk 2]
```diff
@@ -43,31 +42,13 @@ const {
...
-    clientId =
-      getDestinationExternalID(message, 'ga4ClientId') ||
-      get(message, 'anonymousId') ||
-      get(message, 'messageId');
-  }
+const getGA4ClientId = (message) => {
+  const clientId =
+    getDestinationExternalID(message, 'ga4ClientId') ||
+    get(message, 'anonymousId') ||
+    get(message, 'messageId');
```
Comment (ItsSudip) [inline src/v0/destinations/ga4/transform.js:51]: Apart from messageId can't we use rudderId? MessageId doesn't serve the purpose of clientID at all as they are always different for all of the events which can be used for event deduplication but this is not the purpose of clientId isn't it?

Comment (mihir-4116) [reply src/v0/destinations/ga4/transform.js:51]: @ItsSudip this is a fallback option, ideally we are expecting anonymousId only or client_id inside externalId. but we can discuss this option with amy in tomorrow meeting and will post updates here

Comment (shrouti1507) [reply src/v0/destinations/ga4/transform.js:51]: did we discuss on this?

**Pull request closed without merging**